ecs-cli/modules/cli/servicediscovery/servicediscovery_app.go

```diff
@@ -59,26 +62,59 @@ const (
 var requiredParamsSDS = []string{parameterKeyNamespaceID, parameterKeySDSName, parameterKeyDNSType}
 var requiredParamsNamespace = []string{parameterKeyVPCID, parameterKeyNamespaceName}
 
+// Imported Route53 Utility functions that can be mocked in tests
+// Adding the type signature allows code editor autocompletion and checking to work normally
+var findPrivateNamespace route53.FindPrivateNamespaceFunc = route53.FindPrivateNamespace
+var findPublicNamespace route53.FindPublicNamespaceFunc = route53.FindPublicNamespace
+
+// CreateFunc is the interface/signature for Create
+// This helps when writing code in other packages that need to mock this function
```

Comment:
The function type/interface actually isn't needed for tests at all, its just a nicety that makes code auto-complete and error checking work better in IDEs. It actually doesn't add any functionality to the code. If you note above this line, I have:

```
var findPublicNamespace route53.FindPublicNamespaceFunc = route53.FindPublicNamespace
```

In this case `route53.FindPublicNamespaceFunc` is the function type/interface defined in the route53 package. This technically is not needed, because the following would be equivalent from the point of view of the compiler:

```
var findPublicNamespace = route53.FindPublicNamespace
```

But some IDEs (like mine) don't do as well with this (for some reason the normal auto-suggestion/complete features don't work), and so I added the type signatures as a nicety for benefit of future maintainers.

Per your suggestion, I'll move some stuff into a separate helper file.